clippy_lints/src/cfg_not_test.rs

```diff
@@ -0,0 +1,59 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use rustc_ast::NestedMetaItem;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of `cfg` that excludes code from `test` builds. (i.e., `#{cfg(not(test))]`)
+    ///
+    /// ### Why is this bad?
+    /// This may give the false impression that a codebase has 100% coverage, yet actually has untested code.
```

Comment:
I didn't include it here before because I wasn't entirely sure what it meant when writing this, though this should include the 2nd advantage, like
```rust
/// Enabling this also guards against excessive mockery as well, which is an anti-pattern.
```
Could probably be worded better.